runtime/drivers/duckdb/duckdb.go

```diff
@@ -487,6 +493,55 @@ func (c *connection) reopenDB(ctx context.Context) error {
 	}
 	connInitQueries = append(connInitQueries, "SET max_expression_depth TO 250")
 
+	// If AWSRoleARN is provided, assume the role and configure S3 credentials
+	if c.config.AWSRoleARN != "" {
+		var awsCreds *types.Credentials
+		c.awsCredentialsCacheMu.Lock()
+		if c.awsCredentials != nil && c.awsCredentials.Expiration.After(time.Now().Add(5*time.Minute)) {
+			awsCreds = c.awsCredentials
+			c.logger.Info("using cached AWS credentials for S3 access", zap.String("role_arn", c.config.AWSRoleARN), zap.Time("expiration", *awsCreds.Expiration))
+		}
+		c.awsCredentialsCacheMu.Unlock()
+
+		if awsCreds == nil {
+			c.logger.Info("assuming AWS role", zap.String("role_arn", c.config.AWSRoleARN))
+			newCreds, assumeErr := credentials.AssumeRole(ctx, credentials.AssumeRoleOptions{
+				RoleARN:         c.config.AWSRoleARN,
+				RoleSessionName: c.config.AWSRoleSessionName,
+				DurationSeconds: c.config.AWSSessionDurationSeconds,
+			})
+			if assumeErr != nil {
+				return fmt.Errorf("failed to assume AWS role %s: %w", c.config.AWSRoleARN, assumeErr)
+			}
+			awsCreds = newCreds
+			c.logger.Info("assumed AWS role for S3 access", zap.String("role_arn", c.config.AWSRoleARN), zap.Time("expiration", *awsCreds.Expiration))
+
+			// Update cache
+			c.awsCredentialsCacheMu.Lock()
+			c.awsCredentials = awsCreds
+			c.awsCredentialsCacheMu.Unlock()
+		}
+
+		// Set AWS credentials in DuckDB with the conn init queries
+		connInitQueries = append(connInitQueries,
+			fmt.Sprintf("SET s3_access_key_id='%s';", escapeDuckDBString(*awsCreds.AccessKeyId)),
+			fmt.Sprintf("SET s3_secret_access_key='%s';", escapeDuckDBString(*awsCreds.SecretAccessKey)),
+			fmt.Sprintf("SET s3_session_token='%s';", escapeDuckDBString(*awsCreds.SessionToken)))
+
+		if c.config.AWSEndpoint != "" {
+			connInitQueries = append(connInitQueries,
+				fmt.Sprintf("SET s3_endpoint='%s';", escapeDuckDBString(c.config.AWSEndpoint)))
+		}
+		if c.config.AWSRegion != "" {
+			connInitQueries = append(connInitQueries,
+				fmt.Sprintf("SET s3_region='%s';", escapeDuckDBString(c.config.AWSRegion)))
+		}
+		if c.config.AWSS3UseSSL {
+			connInitQueries = append(connInitQueries,
+				fmt.Sprintf("SET s3_use_ssl=%t;", c.config.AWSS3UseSSL))
+		}
```

Comment:
This way of configuring S3 in DuckDB is deprecated – they now recommend using `CREATE SECRET` (e.g. search for `CREATE OR REPLACE TEMPORARY SECRET` in `runtime/drivers/duckdb/model_executor_objectstore_self.go`).

But I'm also a little unsure if we want to be creating/refreshing DuckDB secrets like this – at least if we do, I think it should be explicit not implicit (e.g. something like `create_secrets: [s3]` in the model YAML). I would consider it good enough if we "only" support OIDC for the `objectStoreToSelfExecutor` executor, not managing it for DuckDB's native connectors.

Reply:
The legacy handling better fit the need I hadn't realized it was fully deprecated. I found interacting with the additional manager middleman was awkward. I am fine with just objectStoreToSelfExecutor for the time being